### ISSUE TITLE
Monetization

### ISSUE THREAD

**Issue by Gaelan** [MCModMachine/Design#4]
We need to come up with a solution for both paying our own bills and paying modders and pack devs for their hard work. We need to decide on a strategy for monetization (ads and donations are the obvious options), and be reasonably sure that this strategy will be able to keep the servers running and pay modders competitive rates.

**Comment by DavideRoss** [comment]
We can contact a bunch of MC server hosting companies and ask what they think about a partnership: they provide to pay hosting costs and we provide them ads.

I also thought about a peculiar system for donations to keep the mechanism as transparent as possible: anyone can donate as much they want until reaching the monthly/weekly quota to maintain the service available and be competitive.

Last thing, I thought about premium accounts, I think it's a bad idea because it's not fair that a FOSS product creates disparities between its users.

**Comment by Gaelan** [comment]
I don't see the point in capping donations--all that would accomplish is possibly reducing payouts to modders. It would also be hard to figure out what "competitive" is, because curse does not disclose their rates. We would probably need to do a survey of modders to figure this out. 

I don't see a problem with a premium option they just removed ads or something, but I'm with you on not charging for features.

**Comment by rchard2scout** [comment]
I've seen some discussion on implementing a payment distribution system for modders, where players can say "I'll donate X for this modpack" and the system will distribute that easily and fairly between modders and modpack makers. If we can build something like that, we can take a cut of that.

**Comment by MaPePeR** [comment]
There is NO way you can distribute the amount of money in a modpack in a fair way.

**Comment by Rixx-0** [comment]
You probably should not be trying to pay authors. I would suggest that you focus on Donations to keep the system up and running, and then provide links to the Patreon or other donation link for individual mod authors. Allow the community to donate to the mod makers on their own.

**Comment by Gaelan** [comment]
Very few modders are going to upload their mods to our service as long as Curse is paying them and we aren't.

**Comment by DavideRoss** [comment]
Right now modders are not looking for revenues, but for an alternative platform to Curse, possibly non for-profit, transparent and reliable, and I'm sure that they know that if they will support the project there will be prizes for everyone. I think we have to focus to self-sustainability for now.

**Comment by williewillus** [comment]
Imo an awesome, working, powerful and stable system needs to happen _first_, only then can this happen.

As for the actual topic, I think having a way for players to directly donate to authors would be amazing.

**Comment by Gaelan** [comment]
@williewillus Indeed. First priority is an awesome product. However, [23% of modders list compensation as a "requirement."](https://docs.google.com/forms/d/1e_NaqOyvq9qmDJTUkmCaiVEda61XnG8jedjGagUg_1w/viewanalytics) Compensation will need to come by 1.0.

**Comment by dries007** [comment]
Humble bundle style sliders for donations to modpacks. (setting the defaults would be a different matter)
I also disagree that a platform has to be non profit. I think FOSS but not per say non profit would be a lot better, and more flexible.

**Comment by Gaelan** [comment]
I guess maybe not nonprofit. I want this to be a community project, but, come to think of it, the work done by the developers of this project is just as valuable as the work of anyone else in the community. It's a hard problem.

**Comment by gjgfuj** [comment]
The community is the developers. And vice versa.

**Comment by williewillus** [comment]
finding developers is the least of our worries imo, everyone's willing to contribute if it's FOSS, we just need strong leaders that have high code standards at the top reviewing everything and the platform will be fine.

the way bigger problem that actually needs money is hosting

**Comment by Gaelan** [comment]
Indeed.

**Comment by Gaelan** [comment]
Hosting first, modders and pack creators next, us last.

**Comment by MaPePeR** [comment]
I'm not going to contribute to this project if I know/suspect that the one of us who is doing the money work is filling his own pockets.

**Comment by gjgfuj** [comment]
Clearly We'd need to be completely transparent about it.

**Comment by Gaelan** [comment]
What if we go the Optifine route and offer some nice aesthetic for donating? This could be <del>a generic aesthetic thing implemented by a custom mod added by the client (users might be annoyed by us adding a mod to packs for the sole purpose of getting money, even if the mod itself does nothing), and/or</del> a bundle of mod-specific things implemented using some sort of API.

**Comment by MaPePeR** [comment]
Horrible idea for the same reason you just pointed out. We CANNOT in any way modify any pack or mod.

**Comment by Gaelan** [comment]
@MaPePeR It could always be an API modders could implement to add their own aesthetics (these would have to be disabled by default until players enabled them so that they did not have 10 items from different mods floating above their heads, all z-fighting). This would basically be the same thing as what Optifine and Botania (via Patreon) already do, just consolidated.

**Comment by MaPePeR** [comment]
I doubt anyone would use our service if we would do something like that. That's worse than any other launcher.

**Comment by Gaelan** [comment]
I'll admit I fail to see how it's any different from what e.g. Botania does. Pay a monthly fee (Patreon), get a nice aesthetic thingy. The only difference is that we split the fee instead of having it go to only one mod. However, the worst thing that could happen to this project is getting corrupted by money. If you think something like this would get in the way of making this the best it could be, I'll be the first to toss it out.

**Comment by MaPePeR** [comment]
I think your suggestion falls into the same category as: Show a Watermark with the Logo and let users pay to remove it. The selection of the launcher should not change any pack.

The difference with Botania is: Botania is part of the game content. The launcher is not.

**Comment by Gaelan** [comment]
@MaPePeR I'm not suggesting that we change the pack anymore. That was stupid, I realize that now. Instead, I'm suggesting that basically we provide a centralized interface for what Botania does: player plays money monthly fee, we split this fee between the mods and packs they play, we provide some sort of API to check if a UUID is paying, (e.g.) Botania calls this API and displays a flower over the player's head.
